app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/history/ui/OrderDetails.kt

```diff
@@ -0,0 +1,218 @@
+            OrderItems(order.items, paddingValues)
+        }
+    }
+}
+
+@Composable
+fun OrderHeader(order: Order, showTwoPages: Boolean) {
+    if (showTwoPages) {
+        Text(
+            text = stringResource(R.string.toolbar_history_details_title),
+            style = MaterialTheme.typography.h4,
+            color = MaterialTheme.colors.onBackground
+        )
+        Spacer(Modifier.height(4.dp))
+    }
+
+    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+        OrderDate(orderTimestamp = order.orderTimestamp)
+        Column(verticalArrangement = spacedBy(4.dp), horizontalAlignment = Alignment.End) {
+            OrderId(orderId = order.orderId)
+            OrderAmount(orderPrice = order.totalPrice)
+        }
+    }
+}
+
+@Composable
+fun OrderItems(orderItems: MutableList<OrderItem>, paddingValues: PaddingValues) {
+    LazyColumn(
+        verticalArrangement = spacedBy(25.dp),
+        contentPadding = paddingValues
+    ) {
+        orderItems.map { orderItem ->
+            item {
+                OrderItemPreview(orderItem = orderItem)
+            }
+        }
+    }
+}
+
+@Composable
+fun OrderItemPreview(orderItem: OrderItem) {
+    Box {
+        OrderItemDetails(orderItem = orderItem)
+        OrderItemImage(orderItem = orderItem)
+    }
+}
+
+@Composable
+fun BoxScope.OrderItemDetails(orderItem: OrderItem) {
+    Row(
+        modifier = Modifier
+            .fillMaxSize()
+            .align(Alignment.BottomCenter),
+        horizontalArrangement = spacedBy(41.dp)
+    ) {
+        OrderItemBox()
+        OrderItemText(orderItem = orderItem)
+        OrderItemViewButton(orderItem = orderItem)
+    }
+}
+
+@Composable
+fun RowScope.OrderItemText(orderItem: OrderItem) {
+    Column(modifier = Modifier.weight(5f), verticalArrangement = spacedBy(6.dp)) {
+        Text(
+            text = orderItem.name,
+            style = MaterialTheme.typography.body2,
+            color = MaterialTheme.colors.onBackground
+        )
+        Text(
+            // REVISIT: extract common code from binding adapter formatPrice
+            text = "$" + orderItem.price.toFloat().addThousandsSeparator(),
+            style = MaterialTheme.typography.caption,
+            color = MaterialTheme.colors.onBackground
+        )
+        Text(
+            text = stringResource(R.string.order_quantity, orderItem.quantity),
+            style = MaterialTheme.typography.body1,
+            color = MaterialTheme.colors.onBackground
+        )
+    }
+}
+
+@Composable
+fun RowScope.OrderItemViewButton(orderItem: OrderItem) {
+    TextButton(
+        modifier = Modifier
+            .weight(2f)
+            .align(Alignment.Bottom),
+        shape = CircleShape,
+        colors = ButtonDefaults.buttonColors(
+            backgroundColor = MaterialTheme.colors.primary,
+            contentColor = MaterialTheme.colors.onPrimary
+        ),
+        onClick = { /*TODO*/ }
```

Comment:
Will either try to navigate to the order item in the products tab or the catalog - any suggestions on which would make more sense or suggestions for how to start implementing this?